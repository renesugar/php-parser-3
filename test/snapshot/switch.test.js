const parser = require('../main');

describe("Test SWITCH statements", function() {
  it("parse correctly", function() {
    var ast = parser.parseEval(
      `
      switch(true) {
        case 1:
        case $var:
          $foo = false;
          break 10;
        case FOO:
          $foo = true;
          break;
        default:
          $bar = false;
      }
      switch(true): ?><?php
        case A: ?><?php return true;
        case B: ?><?php return false;
      endswitch;
      `,
      {
        parser: { debug: false }
      }
    );
    expect(ast).toMatchSnapshot();
  });

  it("test errors", function() {
    const errAst = parser.parseEval(
      `
      switch(true);
      switch(true): ?> bug <?php
        case A: return ?><?php true;
        case B: return false  ?><?php ;
      endswitch;
      `,
      {
        parser: { suppressErrors: true }
      }
    );
    expect(errAst).toMatchSnapshot();
    expect(errAst.errors.length).not.toBe(0);
  });
});
