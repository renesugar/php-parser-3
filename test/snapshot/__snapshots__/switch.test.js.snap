// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SWITCH statements parse correctly 1`] = `
Program {
  "children": Array [
    Switch {
      "body": Block {
        "children": Array [
          Case {
            "body": null,
            "kind": "case",
            "test": Number {
              "kind": "number",
              "value": "1",
            },
          },
          Case {
            "body": Block {
              "children": Array [
                ExpressionStatement {
                  "expression": Assign {
                    "kind": "assign",
                    "left": Variable {
                      "byref": false,
                      "curly": false,
                      "kind": "variable",
                      "name": "foo",
                    },
                    "operator": "=",
                    "right": Boolean {
                      "kind": "boolean",
                      "raw": "false",
                      "value": false,
                    },
                  },
                  "kind": "expressionstatement",
                },
                Break {
                  "kind": "break",
                  "level": Number {
                    "kind": "number",
                    "value": "10",
                  },
                },
              ],
              "kind": "block",
            },
            "kind": "case",
            "test": Variable {
              "byref": false,
              "curly": false,
              "kind": "variable",
              "name": "var",
            },
          },
          Case {
            "body": Block {
              "children": Array [
                ExpressionStatement {
                  "expression": Assign {
                    "kind": "assign",
                    "left": Variable {
                      "byref": false,
                      "curly": false,
                      "kind": "variable",
                      "name": "foo",
                    },
                    "operator": "=",
                    "right": Boolean {
                      "kind": "boolean",
                      "raw": "true",
                      "value": true,
                    },
                  },
                  "kind": "expressionstatement",
                },
                Break {
                  "kind": "break",
                  "level": null,
                },
              ],
              "kind": "block",
            },
            "kind": "case",
            "test": Identifier {
              "kind": "identifier",
              "name": ClassReference {
                "kind": "classreference",
                "name": "FOO",
                "resolution": "uqn",
              },
            },
          },
          Case {
            "body": Block {
              "children": Array [
                ExpressionStatement {
                  "expression": Assign {
                    "kind": "assign",
                    "left": Variable {
                      "byref": false,
                      "curly": false,
                      "kind": "variable",
                      "name": "bar",
                    },
                    "operator": "=",
                    "right": Boolean {
                      "kind": "boolean",
                      "raw": "false",
                      "value": false,
                    },
                  },
                  "kind": "expressionstatement",
                },
              ],
              "kind": "block",
            },
            "kind": "case",
            "test": null,
          },
        ],
        "kind": "block",
      },
      "kind": "switch",
      "shortForm": false,
      "test": Boolean {
        "kind": "boolean",
        "raw": "true",
        "value": true,
      },
    },
    Switch {
      "body": Block {
        "children": Array [
          Case {
            "body": Block {
              "children": Array [
                Return {
                  "expr": Boolean {
                    "kind": "boolean",
                    "raw": "true",
                    "value": true,
                  },
                  "kind": "return",
                },
              ],
              "kind": "block",
            },
            "kind": "case",
            "test": Identifier {
              "kind": "identifier",
              "name": ClassReference {
                "kind": "classreference",
                "name": "A",
                "resolution": "uqn",
              },
            },
          },
          Case {
            "body": Block {
              "children": Array [
                Return {
                  "expr": Boolean {
                    "kind": "boolean",
                    "raw": "false",
                    "value": false,
                  },
                  "kind": "return",
                },
              ],
              "kind": "block",
            },
            "kind": "case",
            "test": Identifier {
              "kind": "identifier",
              "name": ClassReference {
                "kind": "classreference",
                "name": "B",
                "resolution": "uqn",
              },
            },
          },
        ],
        "kind": "block",
      },
      "kind": "switch",
      "shortForm": true,
      "test": Boolean {
        "kind": "boolean",
        "raw": "true",
        "value": true,
      },
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`Test SWITCH statements test errors 1`] = `
Program {
  "children": Array [
    Switch {
      "body": Block {
        "children": Array [
          Case {
            "body": Block {
              "children": Array [
                Switch {
                  "body": Block {
                    "children": Array [
                      Case {
                        "body": Block {
                          "children": Array [
                            Inline {
                              "kind": "inline",
                              "raw": " bug ",
                              "value": " bug ",
                            },
                          ],
                          "kind": "block",
                        },
                        "kind": "case",
                        "test": null,
                      },
                      Case {
                        "body": Block {
                          "children": Array [
                            Return {
                              "expr": null,
                              "kind": "return",
                            },
                            ExpressionStatement {
                              "expression": Boolean {
                                "kind": "boolean",
                                "raw": "true",
                                "value": true,
                              },
                              "kind": "expressionstatement",
                            },
                          ],
                          "kind": "block",
                        },
                        "kind": "case",
                        "test": Identifier {
                          "kind": "identifier",
                          "name": ClassReference {
                            "kind": "classreference",
                            "name": "A",
                            "resolution": "uqn",
                          },
                        },
                      },
                      Case {
                        "body": Block {
                          "children": Array [
                            Return {
                              "expr": Boolean {
                                "kind": "boolean",
                                "raw": "false",
                                "value": false,
                              },
                              "kind": "return",
                            },
                          ],
                          "kind": "block",
                        },
                        "kind": "case",
                        "test": Identifier {
                          "kind": "identifier",
                          "name": ClassReference {
                            "kind": "classreference",
                            "name": "B",
                            "resolution": "uqn",
                          },
                        },
                      },
                    ],
                    "kind": "block",
                  },
                  "kind": "switch",
                  "shortForm": true,
                  "test": Boolean {
                    "kind": "boolean",
                    "raw": "true",
                    "value": true,
                  },
                },
              ],
              "kind": "block",
            },
            "kind": "case",
            "test": null,
          },
        ],
        "kind": "block",
      },
      "kind": "switch",
      "shortForm": false,
      "test": Boolean {
        "kind": "boolean",
        "raw": "true",
        "value": true,
      },
    },
  ],
  "errors": Array [
    Error {
      "expected": Array [
        "{",
        ":",
      ],
      "kind": "error",
      "line": 2,
      "message": "Parse Error : syntax error, unexpected ';' on line 2",
      "token": "';'",
    },
    Error {
      "expected": Array [
        204,
        205,
      ],
      "kind": "error",
      "line": 3,
      "message": "Parse Error : syntax error, unexpected 'switch' (T_SWITCH) on line 3",
      "token": "'switch' (T_SWITCH)",
    },
    Error {
      "expected": Array [
        ":",
        ";",
      ],
      "kind": "error",
      "line": 3,
      "message": "Parse Error : syntax error, unexpected 'switch' (T_SWITCH) on line 3",
      "token": "'switch' (T_SWITCH)",
    },
    Error {
      "expected": Array [
        204,
        205,
      ],
      "kind": "error",
      "line": 3,
      "message": "Parse Error : syntax error, unexpected ' bug ' (T_INLINE_HTML) on line 3",
      "token": "' bug ' (T_INLINE_HTML)",
    },
    Error {
      "expected": Array [
        ":",
        ";",
      ],
      "kind": "error",
      "line": 3,
      "message": "Parse Error : syntax error, unexpected ' bug ' (T_INLINE_HTML) on line 3",
      "token": "' bug ' (T_INLINE_HTML)",
    },
    Error {
      "expected": null,
      "kind": "error",
      "line": 7,
      "message": "Parse Error : syntax error on line 7",
      "token": "the end of file (EOF)",
    },
  ],
  "kind": "program",
}
`;
